From 92035bb2924846ea96aeeec9f9d76839c146eaca Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 5 Jun 2018 19:09:43 +0100
Subject: [PATCH] v2v: -o rhv-upload: Set inactivity timeout (RHBZ#1586198).
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This increases the inactivity timeout for transfers from the default
(60 seconds in oVirt < 4.2.3, 600 seconds in >= 4.2.3), up to 1 hour,
so that we should never hit it for ordinary transfers.

Note this requires oVirt >= 4.2.3.  The corresponding oVirt fix was
https://bugzilla.redhat.com/1563278

I also replaced the deprecated ‘image’ parameter with ‘disk’.

Thanks: Nir Soffer, Daniel Erez.
(cherry picked from commit f25404c65f8e078a3ca5bd5a1ab91343edd22506)
---
 v2v/rhv-upload-plugin.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/v2v/rhv-upload-plugin.py b/v2v/rhv-upload-plugin.py
index 791c9e7d2..b4557b83c 100644
--- a/v2v/rhv-upload-plugin.py
+++ b/v2v/rhv-upload-plugin.py
@@ -123,9 +123,8 @@ def open(readonly):
     # Create a new image transfer.
     transfer = transfers_service.add(
         types.ImageTransfer(
-            image = types.Image(
-                id = disk.id
-            )
+            disk = types.Disk(id = disk.id),
+            inactivity_timeout = 3600,
         )
     )
     debug("transfer.id = %r" % transfer.id)
-- 
2.20.1

